### ISSUE TITLE
Paddle的paddle.distribution.Categorical 支持props输入，接口对齐Torch

### ISSUE THREAD

**Issue by zhiminzhang0830** [PaddlePaddle/Paddle#70943]
### 需求描述 Feature Description

paddle.distribution.Categorical 支持props输入

参考链接：
https://www.paddlepaddle.org.cn/documentation/docs/zh/guides/model_convert/convert_from_pytorch/api_difference/distributions/torch.distributions.categorical.Categorical.html

### 替代实现 Alternatives

目前仿照torch进行替代实现：

    eps = paddle.finfo(q_probs.dtype).eps
    q_logits = paddle.log(x=q_probs.clip(min=eps, max=1 - eps))
    d1 = paddle.distribution.Categorical(logits=q_logits)

**Comment by wangguan1995** [comment]
打错字了，是probs参数
https://pytorch.org/docs/stable/distributions.html#categorical

![Image](https://github.com/user-attachments/assets/cf9ba09c-69bd-45a7-ac1b-21d4082679b9)

**Comment by guhaohao0991** [comment]
结果paddle和torch下使用prob或者log_prob方法，结果还是不一样
![Image](https://github.com/user-attachments/assets/b98906d4-3945-455c-9596-3751cbd9ae8f)

![Image](https://github.com/user-attachments/assets/02a4101c-91d5-46f9-b75e-fd26d87f1954)

![Image](https://github.com/user-attachments/assets/9e924cb9-2a30-4e17-bb43-6edea66c9bc1)

![Image](https://github.com/user-attachments/assets/ab7ceecd-bed9-462a-970b-76bff6d7ce2f)
